File: tracking-program/8087/api/models/ReportData.js
/**
 * SwitchCommand.js
 *
 * @description :: A scheduled switch command.
 */

module.exports = {

  attributes: {

    // The project that this switch command belongs to.
    project: {
      model: 'project',
      required: true
    },

    typeId: {
      type: 'number',
      required: true
    },

    type: {
      type: 'string',
      required: true
    },

    valueType: {
      type: 'string',
      required: true
    },

    period: {
      type: 'string',
      required: true
    },

    description: {
      type: 'string',
    },

    value: {
      type: 'number',
      required: true,
    },


  },

};

File: tracking-program/8087/flask_app/app/models/report_data.py
"""ReportData model - api/models/ReportData.js (from xecobase)"""
from app.extensions import db
from app.models.base import BaseModel


class ReportData(BaseModel):
    __tablename__ = "reportdata"

    typeId = db.Column(db.Integer)
    project = db.Column(db.Integer, db.ForeignKey("project.id"))
    type = db.Column(db.String(255), nullable=False)
    valueType = db.Column(db.String(255), nullable=False)
    period = db.Column(db.String(255), nullable=False)
    description = db.Column(db.String(255))
    value = db.Column(db.Float, nullable=False, default=0)
